apps/website/src/content/docs/tree.mdx

```diff
@@ -1,27 +1,149 @@
+Each level of the tree is composed of multiple `TreeNode` components, which can be used to display the node content. `TreeNode` must be used inside `Tree` component to correctly set node `depth` and `subNodes`.
+
+### Subnode
+
+The tree supports hierarchial data structures where each node can have subnodes, which can be expanded or collapsed. Subnodes allow handling nested data up to any desired depth.
+
+Each object in the `data` array can include an array of sub-item objects (e.g., `subItems`) to represent its children, enabling nested structures.
+
+```jsx
+const treeData = React.useMemo(
+  () => [
+    {
+      id: 'Node-1',
+      label: 'Node 1',
+      subItems: [
+        { id: 'Node-1-1', label: 'Node 1.1' },
+        {
+          id: 'Node-1-2',
+          label: 'Node 1.2',
+          subItems: [
+            { id: 'Node-1-2-1', label: 'Node 1.2.1' },
+            { id: 'Node-1-2-2', label: 'Node 1.2.2' },
+          ],
+        },
+      ],
+    },
+    {
+      id: 'Node-2',
+      label: 'Node 2',
+      subItems: [{ id: 'Node-2-1', label: 'Node 2.1' }],
+    },
+  ],
+  [],
+);
+```
+
+To manage expandable subnodes, a state variable is required to track each node and its expansion state.
+
+The `onExpand` function in each `TreeNode` can be used to update the node's expansion state accordingly. This way, when a user expands or collapses a node, its expansion state is updated and the tree re-renders to reflect whether subnodes are visible.
+
+```jsx
+const onExpand = React.useCallback((nodeId, isExpanded) => {
+  setExpandedNodes((prev) => ({
+    ...prev,
+    [nodeId]: isExpanded,
+  }));
+}, []);
+```
 
-## Size
+For each node, the `getNode` function retrieves the current node's information, including whether it has subnodes or it is currently expanded. The information includes:
 
-There are 2 different sizes available. The default size should suffice for most cases. When a smaller version of the tree is needed, use `size="small"`.
+- The `subNodes` property which holds the array of the child nodes contained in the node.
+- The `hasSubNodes` flag which indicates whether the node has subnodes. This is used to determine if node should be expandable.
+- The `isExpanded` flag which indicates whether the node is expanded to display its subnode(s).
+
+```jsx
+const getNode = React.useCallback(
+  (node) => {
+    return {
+      nodeId: node.id,
+      node: node,
+      subNodes: node.subItems,
+      isExpanded: expandedNodes[node.id],
+      hasSubNodes: node.subItems.length > 0,
+    };
+  },
+  [expandedNodes],
+);
+```
+
+<LiveExample src='Tree.subnodes.jsx'>
+  <AllExamples.TreeSubNodesExample client:load />
+</LiveExample>
+
+### Size
+
+There are two different sizes available. The default size should suffice for most cases. When a smaller version of the tree is needed, use `size="small"`.
 
 <LiveExample src='Tree.small.jsx'>
   <AllExamples.TreeSmallExample client:load />
 </LiveExample>
 
+### Visibility
```

Comment:
"Visibility" is too vague. "Visibility checkbox" would be better.

Reply:
I have changed the title.